### DISCUSSION TITLE
ETA on volume-based navigation for flying agents

### DISCUSSION THREAD

**Discussion by RudyFisher7** [recastnavigation/recastnavigation#693] [Q&A] [answered]
Just wondering about the ETA on this feature, as I have a project that requires this functionality and may go with an alternative solution (or likely try to implement myself as there aren't many libraries that offer it) if the time line is far away.

Honestly, I know a lot of game engine developers still say that this is a niche use case, but now a days, I just don't see that as an accurate statement anymore. Games have gotten complex in the past 5 or so years. Hardware can easily support this kind if functionality now, and a lot of games either already use it or end up faking it due to unavailability of a good implementation (or being too coupled with a navmesh system from the past) in the engine they use.

In short, I would advocate that this feature would be a huge game changer and is in much more demand than most developers of this type of technology think.

Thanks!

**Comment by grahamboree** [comment, marked as answer]
Agreed!  I think it's a great feature and is something I'm hoping someone can add soon.  Because Recast is voxel-based it's not terribly difficult to adapt that voxel representation into a 3d navigation volumes approach.

As for timeline, well there really isn't any official ETA.  If this is something you're passionate about and would like to contribute, I'd be happy to check out a PR! ;)

**Reply by RudyFisher7** [reply]
Cool. Good to know thanks! Appreciate this project and work being done!
